### ISSUE TITLE
App fails to start on heroku

### ISSUE THREAD

**Issue by harlantwood** [willzeng/cograph#637]
I created a fresh heroku app, and pushed the repo there.  The bower install went fine, but `lessc` and `r.js` were not found.  Am I missing something here?  Heroku log output below.  

```
2015-05-13T07:18:35.917247+00:00 app[web.1]: > lessc -x app/assets/stylesheets/general.less > app/assets/stylesheets/general.css; lessc -x app/assets/stylesheets/graph.less > app/assets/stylesheets/graph.css; lessc -x app/assets/stylesheets/index-logged-in.less > app/assets/stylesheets/index-logged-in.css; lessc -x app/assets/stylesheets/index.less > app/assets/stylesheets/index.css; lessc -x app/assets/stylesheets/login.less > app/assets/stylesheets/login.css; lessc -x app/assets/stylesheets/mobile.less > app/assets/stylesheets/mobile.css; lessc -x app/assets/stylesheets/nav.less > app/assets/stylesheets/nav.css; lessc -x app/assets/stylesheets/profile.less > app/assets/stylesheets/profile.css; node_modules/requirejs/bin/r.js -o app/build.js; node_modules/requirejs/bin/r.js -o app/build-css.js; node ./bin/www
2015-05-13T07:18:35.917249+00:00 app[web.1]: 
2015-05-13T07:18:35.937873+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.937918+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.937976+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.938096+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.938185+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.938271+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.938349+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.938424+00:00 app[web.1]: sh: 1: lessc: not found
2015-05-13T07:18:35.938855+00:00 app[web.1]: sh: 1: node_modules/requirejs/bin/r.js: not found
2015-05-13T07:18:35.945095+00:00 app[web.1]: sh: 1: node_modules/requirejs/bin/r.js: not found
2015-05-13T07:18:37.841197+00:00 app[web.1]: npm ERR! Linux 3.13.0-49-generic
2015-05-13T07:18:37.850476+00:00 app[web.1]: npm ERR! Please include the following file with any support request:
2015-05-13T07:18:37.850703+00:00 app[web.1]: npm ERR!     /app/npm-debug.log
2015-05-13T07:18:37.805039+00:00 app[web.1]: /app/node_modules/passport-twitter/node_modules/passport-oauth1/lib/strategy.js:83
2015-05-13T07:18:37.805045+00:00 app[web.1]:   if (!options.consumerKey) { throw new TypeError('OAuthStrategy requires a co
2015-05-13T07:18:37.805047+00:00 app[web.1]:                                     ^
2015-05-13T07:18:37.805048+00:00 app[web.1]: TypeError: OAuthStrategy requires a consumerKey option
2015-05-13T07:18:37.805052+00:00 app[web.1]:     at Strategy.OAuthStrategy (/app/node_modules/passport-twitter/node_modules/passport-oauth1/lib/strategy.js:83:37)
2015-05-13T07:18:37.805053+00:00 app[web.1]:     at new Strategy (/app/node_modules/passport-twitter/lib/strategy.js:53:17)
2015-05-13T07:18:37.805054+00:00 app[web.1]:     at module.exports (/app/config/passport.coffee:49:18)
2015-05-13T07:18:37.805055+00:00 app[web.1]:     at Object.<anonymous> (/app/app.coffee:36:31)
2015-05-13T07:18:37.805057+00:00 app[web.1]:     at Object.<anonymous> (/app/app.coffee:97:4)
2015-05-13T07:18:37.805058+00:00 app[web.1]:     at Module._compile (module.js:460:26)
2015-05-13T07:18:37.805059+00:00 app[web.1]:     at Object.require.extensions..coffee (/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:21:21)
2015-05-13T07:18:37.805060+00:00 app[web.1]:     at Module.load (module.js:355:32)
2015-05-13T07:18:37.805062+00:00 app[web.1]:     at Function.Module._load (module.js:310:12)
2015-05-13T07:18:37.805063+00:00 app[web.1]:     at Module.require (module.js:365:17)
2015-05-13T07:18:37.835951+00:00 app[web.1]: 
2015-05-13T07:18:37.841920+00:00 app[web.1]: npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "start"
2015-05-13T07:18:37.842284+00:00 app[web.1]: npm ERR! node v0.12.2
2015-05-13T07:18:37.843113+00:00 app[web.1]: npm ERR! npm  v2.7.4
2015-05-13T07:18:37.843430+00:00 app[web.1]: npm ERR! code ELIFECYCLE
2015-05-13T07:18:37.843731+00:00 app[web.1]: npm ERR! graphdocs@0.0.0 start: `lessc -x app/assets/stylesheets/general.less > app/assets/stylesheets/general.css; lessc -x app/assets/stylesheets/graph.less > app/assets/stylesheets/graph.css; lessc -x app/assets/stylesheets/index-logged-in.less > app/assets/stylesheets/index-logged-in.css; lessc -x app/assets/stylesheets/index.less > app/assets/stylesheets/index.css; lessc -x app/assets/stylesheets/login.less > app/assets/stylesheets/login.css; lessc -x app/assets/stylesheets/mobile.less > app/assets/stylesheets/mobile.css; lessc -x app/assets/stylesheets/nav.less > app/assets/stylesheets/nav.css; lessc -x app/assets/stylesheets/profile.less > app/assets/stylesheets/profile.css; node_modules/requirejs/bin/r.js -o app/build.js; node_modules/requirejs/bin/r.js -o app/build-css.js; node ./bin/www`
2015-05-13T07:18:37.844006+00:00 app[web.1]: npm ERR! Exit status 1
2015-05-13T07:18:37.844269+00:00 app[web.1]: npm ERR! 
2015-05-13T07:18:37.844535+00:00 app[web.1]: npm ERR! Failed at the graphdocs@0.0.0 start script 'lessc -x app/assets/stylesheets/general.less > app/assets/stylesheets/general.css; lessc -x app/assets/stylesheets/graph.less > app/assets/stylesheets/graph.css; lessc -x app/assets/stylesheets/index-logged-in.less > app/assets/stylesheets/index-logged-in.css; lessc -x app/assets/stylesheets/index.less > app/assets/stylesheets/index.css; lessc -x app/assets/stylesheets/login.less > app/assets/stylesheets/login.css; lessc -x app/assets/stylesheets/mobile.less > app/assets/stylesheets/mobile.css; lessc -x app/assets/stylesheets/nav.less > app/assets/stylesheets/nav.css; lessc -x app/assets/stylesheets/profile.less > app/assets/stylesheets/profile.css; node_modules/requirejs/bin/r.js -o app/build.js; node_modules/requirejs/bin/r.js -o app/build-css.js; node ./bin/www'.
2015-05-13T07:18:37.844869+00:00 app[web.1]: npm ERR! This is most likely a problem with the graphdocs package,
2015-05-13T07:18:37.845120+00:00 app[web.1]: npm ERR! not with npm itself.
2015-05-13T07:18:37.845368+00:00 app[web.1]: npm ERR! Tell the author that this fails on your system:
2015-05-13T07:18:37.846073+00:00 app[web.1]: npm ERR!     lessc -x app/assets/stylesheets/general.less > app/assets/stylesheets/general.css; lessc -x app/assets/stylesheets/graph.less > app/assets/stylesheets/graph.css; lessc -x app/assets/stylesheets/index-logged-in.less > app/assets/stylesheets/index-logged-in.css; lessc -x app/assets/stylesheets/index.less > app/assets/stylesheets/index.css; lessc -x app/assets/stylesheets/login.less > app/assets/stylesheets/login.css; lessc -x app/assets/stylesheets/mobile.less > app/assets/stylesheets/mobile.css; lessc -x app/assets/stylesheets/nav.less > app/assets/stylesheets/nav.css; lessc -x app/assets/stylesheets/profile.less > app/assets/stylesheets/profile.css; node_modules/requirejs/bin/r.js -o app/build.js; node_modules/requirejs/bin/r.js -o app/build-css.js; node ./bin/www
2015-05-13T07:18:37.846374+00:00 app[web.1]: npm ERR! You can get their info via:
2015-05-13T07:18:37.846610+00:00 app[web.1]: npm ERR!     npm owner ls graphdocs
2015-05-13T07:18:37.846851+00:00 app[web.1]: npm ERR! There is likely additional logging output above.
2015-05-13T07:18:37.849855+00:00 app[web.1]: 
2015-05-13T07:18:38.829670+00:00 heroku[web.1]: Process exited with status 1
2015-05-13T07:18:38.845114+00:00 heroku[web.1]: State changed from starting to crashed
```

**Comment by harlantwood** [comment]
The `lessc` issue is fixed by #638 -- still not sure why the `r.js` issue is happening...

**Comment by harlantwood** [comment]
Should be all fixed by #639.
